3_callbacks/node_4.x/addon.cc

```diff
@@ -0,0 +1,18 @@
+#include <node.h>
+
+using namespace v8;
+
+void RunCallback(const FunctionCallbackInfo<Value>& args) {
+  Isolate* isolate = args.GetIsolate();
+
+  Local<Function> cb = Local<Function>::Cast(args[0]);
```

Comment:
Without looking at the surrounding context, I'd say yes. However, the default return value is set to Undefined, so a simple return statement would suffice.

On February 5, 2016 12:23:39 PM GMT+02:00, Tom Gallacher <notifications@github.com> wrote:
>> @@ -0,0 +1,18 @@
>> +#include <node.h>
>> +
>> +using namespace v8;
>> +
>> +void RunCallback(const FunctionCallbackInfo<Value>& args) {
>> +  Isolate* isolate = args.GetIsolate();
>> +
>> +  Local<Function> cb = Local<Function>::Cast(args[0]);
>
>Would the following make sense?
>```c++
>if (!args[0]->IsFunction()) {
>  return args.GetReturnValue().SetUndefined();
>}
>```
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/node-addon-examples/pull/49/files#r51998512